server/src/main/java/org/elasticsearch/index/codec/vectors/es818/ES818FlatVectorsFormat.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * @notice
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Modifications copyright (C) 2024 Elasticsearch B.V.
+ */
+package org.elasticsearch.index.codec.vectors.es818;
+
+import org.apache.lucene.codecs.hnsw.FlatVectorsFormat;
+import org.apache.lucene.codecs.hnsw.FlatVectorsReader;
+import org.apache.lucene.codecs.hnsw.FlatVectorsScorer;
+import org.apache.lucene.codecs.hnsw.FlatVectorsWriter;
+import org.apache.lucene.codecs.lucene99.Lucene99FlatVectorsWriter;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.index.SegmentWriteState;
+
+import java.io.IOException;
+
+/**
+ * Copied from Lucene99FlatVectorsFormat in Lucene 10.1
+ *
+ * This is copied to change the implementation of {@link #fieldsReader} only.
+ * The codec format itself is not changed, so we keep the original {@link #NAME}
+ */
+public class ES818FlatVectorsFormat extends FlatVectorsFormat {
```

Comment:
nit: should it be 819?

Reply:
The backport will be to 8.19 at best, introducing a new behaviour in 8.18.x is not really an option.

Reply:
How do you expect the various interrelated classes to interact? I can move this to `vectors.es910.ES910FlatVectorsFormat`, that will be referenced by `ES818FlatVectorsFormat`? Or should the whole lot move to `es910` package?

Reply:
We don't really introduce a new format here. This is just a wrapper so maybe we can simply remove the ES818 from the name? 
It should be called something like DirectIOFlatVectorsFormatWrapper.

Reply:
I've tweaked the names to not be version-specific